health.go

```diff
@@ -242,25 +286,57 @@ func (h *Health) startRunner(cfg *Config, ticker *time.Ticker, stop <-chan struc
 
 		h.Logger.WithFields(log.Fields{"name": cfg.Name}).Debug("Checker exiting")
 	}()
-
-	return nil
 }
 
+// resets the states in a concurrency-safe manner
 func (h *Health) safeResetStates() {
 	h.statesLock.Lock()
 	defer h.statesLock.Unlock()
 	h.states = make(map[string]State, 0)
 }
 
+// updates the check state in a concurrency-safe manner
 func (h *Health) safeUpdateState(stateEntry *State) {
 	// update states here
 	h.statesLock.Lock()
 	defer h.statesLock.Unlock()
+
+	// dispatch any status listeners
+	h.handleStatusListener(stateEntry)
```

Comment:
could we get away with doing this outside the lock?

Reply:
ahh great point.  Yeah there's no reason this needs to be inside the lock

Reply:
^ I think this does need to be inside the lock since handleStatusListener does this:

```
...
	// get the previous state
	prevState := h.states[stateEntry.Name]
...
```

Reply:
grrr.  Correct-O